File: 0696-count-binary-substrings/0696-count-binary-substrings.py
class Solution:
    def countBinarySubstrings(self, s: str) -> int:
        count = 0
        prev_count = 0
        curr_count = 1

        for i in range(1, len(s)):
            if s[i] == s[i-1]:
                curr_count += 1
            else:
                prev_count = curr_count
                curr_count = 1

            if prev_count >= curr_count:
                  count += 1

        return count

File: 0696-count-binary-substrings/0696-count-binary-substrings.cpp
class Solution {
public:
    int countBinarySubstrings(string s) {
     int n = s.size();
     int prev =0;
     int curr=1;
     int answer=0;   
     for(int i=1;i<n;i++){
         if(s[i]==s[i-1]){
             curr++;
         }
         else{
             answer+=min(prev,curr);
             prev=curr;
             curr=1;
            }
           
         } 
       
      answer+=min(prev,curr);
     return answer;   
    }
};